### PR TITLE
Resolve versionless dependencies #78

### PR DESCRIPTION
**Pull request by TG1999**
Signed-off-by: Tushar Goel <tushar.goel.dav@gmail.com>

### CODE REVIEW

File: tests/test_cli.py [hunk 1]
```diff
@@ -162,6 +162,24 @@ def test_cli_with_environment_marker_and_complex_ranges():
...
 
+@pytest.mark.online
+def test_cli_with_azure_devops():
+    requirements_file = test_env.get_test_loc("azure-devops.req.txt")
+    expected_file = test_env.get_test_loc("azure-devops.req-expected.json", must_exist=False)
+    extra_options = [
+        "--operating-system",
+        "linux",
+        "--python-version",
+        "38",
```
Comment (pombredanne) [inline tests/test_cli.py:173]: Can you make another tests using 310?

Comment (TG1999) [reply tests/test_cli.py:173]: Sure!

File: src/python_inspector/resolution.py [hunk 2]
```diff
@@ -192,10 +192,11 @@ def is_valid_version(
     """
     Return True if the parsed_version is valid for the given identifier.
     """
-    if (
-        any(parsed_version not in r.specifier for r in requirements[identifier])
-        or parsed_version in bad_versions
-    ):
+    if parsed_version in bad_versions:
```
Comment (pombredanne) [inline src/python_inspector/resolution.py:195]: Do you mind to add unit tests for this `is_valid_version` function?

File: src/python_inspector/resolution.py [hunk 3]
```diff
@@ -453,16 +454,22 @@ def get_candidates(
...
+            if is_valid_version(
                 parsed_version=parsed_version,
                 requirements=requirements,
                 identifier=identifier,
                 bad_versions=bad_versions,
             ):
-                continue
-            yield Candidate(name=name, version=parsed_version, extras=extras)
+                valid_versions.append(parsed_version)
+        if all(version.is_prerelease for version in valid_versions):
```
Comment (pombredanne) [inline src/python_inspector/resolution.py:467]: What about removing the pass below and using this instead?

```suggestion
        if not all(version.is_prerelease for version in valid_versions):
```

or may be there is a way to better express your intentions?

**Review by pombredanne — COMMENTED** [review]
Thanks!
Can you please also add a changelog entry and prepare a new dot release in this same PR?

**Review by pombredanne — APPROVED** [review]
LGTM!
Thanks

**Pull request merged**